wrapanapi/systems/google.py

```diff
@@ -0,0 +1,798 @@
+        GCE service is stateless, so there's nothing to disconnect from
+        """
+        pass
+
+    def list_vms(self, zones=None):
+        """
+        List all VMs in the GCE account, filtered by zone if desired
+
+        Args:
+            zone: List of zones, by default does not filter to any zone
+
+        Returns:
+            List of GCEInstance objects
+        """
+        results = []
+        if not zones:
+            zones = [
+                zone['name'].split('/')[-1] for zone  # convert url-based name
+                in self._compute.zones().list(project=self._project).execute().get('items', [])
+            ]
+
+        for zone_name in zones:
+            zone_instances = self._instances.list(
+                project=self._project, zone=zone_name).execute()
+            for instance in zone_instances.get('items', []):
+                results.append(
+                    GCEInstance(
+                        system=self, raw=instance, name=instance['name'], zone=zone_name
+                    )
+                )
+
+        return results
+
+    def find_vms(self, name, zones=None):
+        """
+        Find VMs with a given name, filtered by zones if desired
+
+        Args:
+            zones: List of zone names, by default does not filter to any zone
+        Returns:
+            List of GCEInstance objects that match
+        """
+        results = []
+        if not zones:
+            zones = [
+                zone['name'].split('/')[-1] for zone  # convert url-based name
+                in self._compute.zones().list(project=self._project).execute().get('items', [])
+            ]
+
+        for zone_name in zones:
+            try:
+                # Just use get in each zone instead of iterating through all instances
+                instance = self._instances.get(
+                    project=self._project, zone=zone_name, instance=name).execute()
+                results.append(
+                    GCEInstance(system=self, raw=instance, name=instance['name'], zone=zone_name))
+            except errors.HttpError as error:
+                if error.resp.status == 404:
+                    # Getting the instance we found in the list failed, just skip it...
+                    pass
+                else:
+                    raise
+        return results
+
+    def get_vm(self, name, zone=None):
+        """
+        Get a single VM with given name in a specified zone
+
+        By default self._zone is used
+
+        Args:
+            zone: zone to get VM from, defaults to self._zone
+        Returns:
+            GCEInstance object
+        Raises:
+            VMInstanceNotFound if unable to find vm
+            MultipleInstancesError if multiple vm's with the same name found
+        """
+        if not zone:
+            zone = self._zone
+        instances = self.find_vms(name, zones=[zone])
+        if not instances:
+            self.logger.info("Looking for instance '%s' in all zones", name)
```

Comment:
I don't think this is particularly good behavior, given that there's no way for the caller to know that this happened, they just get a vm back, that might not be the one that they actually wanted from their given zone, which they'll take action on.

This should just raise `VMInstanceNotFound` at this point, since the VM was not in the zone.

Reply:
This logic existed before the refactor. I'll make this behavior a kwarg option, disabled by default.